### PR TITLE
Refactor JSON-RPC code - test_rpc_endpoints.py

### PR DESCRIPTION
**Pull request by drknzz**
## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [ ] No linter errors
- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: test/rpc/conftest.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"""
+Tests RPC endpoints.
+"""
```
Comment (cptartur) [inline test/rpc/conftest.py:3]: This docstring doesn't make much sense in this file

File: test/rpc/conftest.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+from test.util import load_file_content
+from test.test_endpoints import send_transaction
+
+from test_rpc_endpoints import gateway_call
+
+
+DEPLOY_CONTENT = load_file_content("deploy_rpc.json")
+INVOKE_CONTENT = load_file_content("invoke_rpc.json")
+DECLARE_CONTENT = load_file_content("declare.json")
```
Comment (cptartur) [inline test/rpc/conftest.py:22]: Have you verified if these work correctly now that rpc test are in a separate directory?

File: test/rpc/conftest.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+@pytest.fixture(name="declare_info", scope="module")
+def fixture_declare_info() -> dict:
+    """
+    Make a declare transaction on devnet and return declare info dict
+    """
+    declare_tx = json.loads(DECLARE_CONTENT)
+    resp = send_transaction(declare_tx)
+    declare_info = json.loads(resp.data.decode("utf-8"))
+    return {**declare_info, **declare_tx}
```
Comment (cptartur) [inline test/rpc/conftest.py:73]: Empty line missing at the end of the file

File: test/rpc/test_rpc_blocks.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
+"""
+Tests RPC blocks.
```
Comment (cptartur) [inline test/rpc/test_rpc_blocks.py:2]:
```suggestion
Tests RPC blocks
```

File: test/rpc/test_rpc_endpoints.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+def get_block_with_transaction(transaction_hash: str) -> dict:
+    """
+    Retrieve block for given transaction
+    """
+    transaction = gateway_call("get_transaction", transactionHash=transaction_hash)
+    block_number: int = transaction["block_number"]
+    block = gateway_call("get_block", blockNumber=block_number)
+    return block
+
```
Comment (cptartur) [inline test/rpc/test_rpc_endpoints.py:55]: I'd move these methods into separate, something like common/utils/whatever

File: test/rpc/test_rpc_storage.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
+"""
+Tests RPC storage.
```
Comment (cptartur) [inline test/rpc/test_rpc_storage.py:2]:
```suggestion
Tests RPC storage
```

**Review by cptartur — CHANGES_REQUESTED** [review]
Couple of suggestions

File: test/rpc/conftest.py (hunk 2 above)
Comment (drknzz) [reply test/rpc/conftest.py:22]: Yes, the function uses path relative to the file it's defined in:

https://github.com/drknzz/starknet-devnet/blob/0501541a0772b082faf65030adab2cc95bd25f41/test/util.py#L410-L414

File: test/rpc/test_rpc_class.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 Tests RPC contract class.
```
Comment (cptartur) [inline test/rpc/test_rpc_class.py:2]:
```suggestion
Tests RPC contract class
```

Comment (cptartur) [inline test/rpc/test_rpc_class.py:2]:
```suggestion
Tests RPC contract class
```

**Review by cptartur — COMMENTED** [review]
Just a single overlooked full stop
Other than that, good work 👍

**Pull request closed without merging**